Shopify/cli
# Make developer console support create/delete extensions mid-dev

### WHY are these changes introduced?

Adds support for handling extension creation and deletion events in the developer console during consistent-dev.

Fixes https://github.com/Shopify/develop-app-outer-loop/issues/1198

### WHAT is this pull request doing?

- Implements handling for Created and Deleted extension events
- Adds new methods to ExtensionsPayloadStore for adding and removing extensions
- Updates the event handler to maintain the extensions list in the payload store options

KNOWN ISSUES:
- When removing an extension, you need to manually refresh the dev-console.

### How to test your changes?

1. Start dev for an app with UI extensions
2. Create a new UI extension and verify it appears in the dev console (and that the deeplink works)
3. Delete a UI extension and verify it's removed from the dev console (after refreshing)
4. Modify an existing extension and confirm updates are still working

### Measuring impact


### Checklist
